src/JakobSteinn/HTMLFormValidator/Validator.php

```diff
@@ -0,0 +1,174 @@
+<?php
+namespace JakobSteinn\HTMLFormValidator;
+
+use \DOMElement;
+use \DOMDocument;
+use JakobSteinn\HTMLFormValidator\RuleProviders\URLRuleProvider;
+use JakobSteinn\HTMLFormValidator\RuleProviders\EmailRuleProvider;
+use JakobSteinn\HTMLFormValidator\Exceptions\UnknownValidationRule;
+
+class Validator
+{
+    /**
+     * The document that will be searched for a form
+     * 
+     * @var DOMDocument
+     */
+    protected $formDocument;
+
+    /**
+     * The rule providers used to validate input
+     * 
+     * @var JakobSteinn\HTMLFormValidator\RuleProviders\AbstractRuleProvider
+     */
+    protected $ruleProviders = [
```

Comment:
This is indeed what the [`ValidatorPluginManager`](https://github.com/zendframework/zend-validator/blob/33f0141fce05155fb75b5d68fac5856d3a21ebd2/src/ValidatorPluginManager.php#L32) does in zf2 (sorry if I always use zf2 as an example, it's just that that is my area of expertise)